PURL-TYPES.rst

```diff
@@ -106,8 +106,21 @@ conda
 
       pkg:conda/absl-py@0.4.1?build=py36h06a4308_0&channel=main&subdir=linux-64&type=tar.bz2
 
+cpan
```

Comment:
this is a new definition

Reply:
CPAN was already defined in PURL spec (#288)

Reply:
oh, this was not clear in this outdated PR ;-) 
@gfs could you rebase on latest master and fix the conflicts?

Reply:
I've resolve the conflicts. It looks like CPAN has been merged via another PR in the interim while this was open. Now this PR only fixes alphabetization (cargo comes before cocoapods) and adds vsm. Let me know if there's anything else you'd need from me, thanks.